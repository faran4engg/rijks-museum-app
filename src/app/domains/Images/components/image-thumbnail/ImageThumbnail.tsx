import { useEffect, useState } from 'react';
import { ImageThumbnailProps } from './types';
import placeholder from 'assets/images/placeholder.gif';
import ImageDetails from '../image-details/ImageDetails';

const ImageThumbnail: React.FC<ImageThumbnailProps> = ({
  url,
  ...imgDetailsProps
}) => {
  const [placeHolder, setPlaceHolder] = useState(placeholder);
  const [showDetails, setShowDetails] = useState(false);

  useEffect(() => {
    setTimeout(() => {
      setShowDetails(true);
    }, 1500);
  });

  return (
    <div className="flex flex-col items-center justify-center img-grid-item w-full">
      <img
        loading="lazy"
        onLoad={() => {
          setPlaceHolder(url);
        }}
        src={placeHolder}
        alt="painting-art"
        className="w-full bg-gray-300 bg-center bg-auto rounded-lg shadow-md"
      />
      {showDetails && <ImageDetails {...imgDetailsProps} />}
    </div>
  );
};

export default ImageThumbnail;
