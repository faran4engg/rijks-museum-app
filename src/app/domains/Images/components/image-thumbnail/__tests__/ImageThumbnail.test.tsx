import { create } from 'react-test-renderer';

import ImageThumbnail from '../ImageThumbnail';

describe('ImageThumbnail', () => {
  it('should render', () => {
    const component = create(
      <ImageThumbnail authorName="faran" title="my painting" url="test.com" />,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
