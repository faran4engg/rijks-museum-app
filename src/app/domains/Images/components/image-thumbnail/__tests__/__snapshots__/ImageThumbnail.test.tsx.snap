// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageThumbnail should render 1`] = `
<div
  className="flex flex-col items-center justify-center img-grid-item w-full"
>
  <img
    alt="painting-art"
    className="w-full bg-gray-300 bg-center bg-auto rounded-lg shadow-md"
    loading="lazy"
    onLoad={[Function]}
    src="placeholder.gif"
  />
</div>
`;
